modules/storages/lib/api/v3/file_links/file_links_download_api.rb

```diff
@@ -28,15 +28,40 @@
 
 class API::V3::FileLinks::FileLinksDownloadAPI < ::API::OpenProjectAPI
   helpers Storages::Peripherals::StorageUrlHelper
+  using Storages::Peripherals::ServiceResultRefinements
+
+  helpers do
+    def raise_error(error)
+      case error
+      when :not_found
+        raise API::Errors::NotFound.new
+      when :not_authorized
+        raise API::Errors::Unauthorized.new
```

Comment:
I just wondered, why this error was not part of the previous code. And then I realized, that this code now will return 403, if the request to NC is unauthorized. I'm pretty sure, this is an effect, we do not want, because it can lead to some misunderstandings. the authorization to NC is something, that is handled by server code, not the user having outdated sessions or else. If an API returns 403, I'd suspect, it means that my authorization with this specific server is expired or missing, or else.

what do you think?

Reply:
Yes I agree, what shall we return instead? 500?

Reply:
Yes, I think so. Because we are having an error in backend code execution basically.